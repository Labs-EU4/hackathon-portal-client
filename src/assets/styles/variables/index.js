export { Solid, Gradient } from './colors';
export {
    type,
    baseFontSize,
    smallFontSize,
    h1FontSize,
    h2FontSize,
    h3FontSize,
    h4FontSize,
    button_text
} from './fonts';
export { Spacing } from './spacing';
export { media } from './media';