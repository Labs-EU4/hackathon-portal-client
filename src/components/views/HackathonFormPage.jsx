import React from "react";
import { useParams } from "react-router-dom";
import { useSelector } from "react-redux";

import HackathonForm from "../templates/HackathonForm";

const HackathonFormPage = () => {
  const { id } = useParams();
  const event = useSelector(state =>
    state.events.data.find(e => e.id === Number(id))
  );

  return <HackathonForm initialState={event} />;
};

export default HackathonFormPage;