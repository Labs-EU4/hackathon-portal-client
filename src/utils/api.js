import Axios from "axios";
import { toast } from "react-toastify";
import { resetUser } from "../store/user/actions";

const baseURL = process.env.REACT_APP_API_URL;

// if process.env.CONTEXT = "production" {
//   Point to heroku API
// }
// if process.env.BRANCH = "develop" {
  // point to Heroku staging API
// }
// if process.env.CONTEXT = "deploy-preview" {
//   point to deploy-preview HEROKU API (the separate one)
// }



export const axiosWithAuth = token =>
  Axios.create({
    baseURL,
    headers: {
      Authorization: token
    }
  });

export const axios = Axios.create({ baseURL });

export const selectToken = state => state.currentUser.token;

export const showError = message => {
  toast.error(message);
};

export const showSuccess = message => {
  toast.success(message);
};

export function* handleError({ response }, put, history) {
  const { message, statusCode } = response.data;
  if (statusCode === 404) {
    history.push("/not-found");
  }
  if (message === "jwt expired") {
    yield put(resetUser());
  } else showError(`⚠️ ${message}`);
}
