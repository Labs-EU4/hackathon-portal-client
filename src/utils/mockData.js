export const initialState = {
  events: {
    data: [
      {
        id: 1,
        event_title: "EuroHack",
        event_description: "the best event hack hack hack macbook",
        creator_id: 1,
        start_date: "10/09/11",
        end_date: "08/07/13",
        location: "Europe",
        guidelines: "cool hekhkdsjfhsdkjf skdjfhskdjfhkjsdfh skjdfhksjdfh",
        participation_type: "Team",
        category_id: 1,
        requirements: [
          {
            video_url:
              "https://res.cloudinary.com/ibkrice-com/video/upload/v1581259780/Harvesting_ifato8.mp4"
          }
        ],
        ProfileImage: [
          {
            image_url:
              "https://res.cloudinary.com/ibkrice-com/video/upload/v1581259780/Harvesting_ifato8.mp4"
          }
        ],
        rubrics: [
          "Presentation",
          "Product Fit",
          "Functionality",
          "Innovation",
          "Product Design",
          "Extensibility"
        ]
      },
      {
        id: 2,
        event_title: "World",
        event_description: "number one hacker event in the world",
        creator_id: 1,
        start_date: "07/09/11",
        end_date: "08/07/13",
        location: "Asia",
        guidelines: "come well prepared,no cheating,no fighting",
        participation_type: "Team",
        category_id: 2
      }
    ],
    categories: [
      { id: 2, category_name: "Summer Hackton" },
      { id: 3, category_name: "World Hackton" },
      { id: 3, category_name: "Asia Hackton" },
      { id: 3, category_name: "Gamer's Hackton" },
      { id: 3, category_name: "Advanced Hackton" }
    ]
  },

  currentUser: {
    token: 347534574536,
    userId: 1,
    fullname: "Jake Tom",
    email: "8omemail@google.com",
    username: "Jake22",
    bio: "I like coding ",
    image: ""
  },
  submissions: [
    {
      id: 1,
      project_title: "Mock Project",
      participant_or_team_name: "Mock Team",
      event_id: 7,
      video_url: "",
      git_url: "https://www.google.com/",
      project_writeups: "No mock write up",
      submitted_by: 14,
      created_at: "2020-03-02",
      updated_at: "2020-03-02",
      average_rating: 0,
      acted_judges: 0,
      number_of_judges: 0
    }
  ],

  teams: [
    { id: 1, team_name: "The A team", team_lead: 2 },
    { id: 2, team_name: "Hacking central", team_lead: 3 },
    { id: 3, team_name: "blackhats", team_lead: 4 },
    { id: 4, team_name: "Spies", team_lead: 1 }
  ],
  participant_team_members: [
    {
      team_member: 3,
      team_id: 1
    },
    {
      team_member: 4,
      team_id: 1
    },
    {
      team_member: 5,
      team_id: 1
    }
  ]
};
