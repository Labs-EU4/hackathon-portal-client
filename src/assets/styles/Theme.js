import * as color from './variables/colors';

const size = {
    mobileS: '320px',
    mobileM: '375px',
    mobileL: '425px',
    tablet: '768px',
    laptop: '1024px',
    laptopL: '1440px',
    desktop: '2560px'
};

export const theme = {
    color: {
        primary: {
            // light: color.primaryLight,
            regular: color.primary,
            // dark: color.primaryDark
        },
        secondary: {
            // light: color.secondaryLight,
            regular: color.secondary,
            // dark: color.secondaryDark
        },
        white: {
            regular: color.white,
        },
        black: {
            regular: color.black,
        },
        red: {
            regular: color.red,
            bright: color.brightRed
        },
        // grey: color.grey,
        grey: {
            regular: color.grey
        },
        green: {
            light: color.lightGreen,
            regular: color.green,
            dark: color.darkGreen
        },
    },
    shadow: {
        text: `text-shadow: .15px .15px 1px ${color.black}, .75px .5px 1.5px ${color.black}, .15px .15px 1px ${color.black};`,
        text_links: `1px .75px 1.5px ${color.black}`,
        box: `box-shadow: 0 3px 20px 0 rgba(0, 0, 0, .2)`,
        card: `box-shadow: 1px .5px 3px rgba(0, 0, 0, .15)`,
        intenseBox: `box-shadow: 0 2.5px 10px ${color.black};`,
        filter: `filter: drop-shadow(.5px .25px 1.5px ${color.black});
        filter: drop-shadow(.15px .15px 1px ${color.black}, .5px .25px 1.5px ${color.black}, .65px .4px 1px ${color.black});`,
        icon: `filter: drop-shadow(.5px .25px 1.5px ${color.black});
        filter: drop-shadow(25px .25px 10px ${color.black}, .5px .25px 1.5px ${color.black}, .65px .4px 1px ${color.black});`,
        btn: `2px 2px 3px ${color.black}`
    },
    flex: {
        center: 'display: flex; justify-content: center; align-items: center;',
        columnCenter: 'display: flex; flex-direction: column; justify-content: center; align-items: center;',
        column: 'display: flex; flex-direction: column;',

        custom: (justify='start', align='start', direction='row') => {
            return `
                display: flex; 
                flex-direction: ${direction}
                justify-content: ${justify}; 
                align-items: ${align};
            `
        },
    },
    device: {
        mobileS: `(min-width: ${size.mobileS})`,
        mobileM: `(min-width: ${size.mobileM})`,
        mobileL: `(min-width: ${size.mobileL})`,
        tablet: `(min-width: ${size.tablet})`,
        laptop: `(min-width: ${size.laptop})`,
        laptopL: `(min-width: ${size.laptopL})`,
        desktop: `(min-width: ${size.desktop})`,
        desktopL: `(min-width: ${size.desktop})`
    }
};