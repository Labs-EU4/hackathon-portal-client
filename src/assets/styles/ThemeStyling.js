import * as color from './variables/colors';

const size = {
    mobileS: '320px',
    mobileM: '375px',
    mobileL: '425px',
    tablet: '768px',
    laptop: '1024px',
    laptopL: '1440px',
    desktop: '2560px'
};

export const theme = {
    color: {
        primary: {
            light: '#81DBFD',
            regular: color.primary,
            // dark: color.primaryDark
        },
        secondary: {
            // light: color.secondaryLight,
            regular: color.secondary,
            // dark: color.secondaryDark
        },
        white: {
            regular: color.white,
            bg: color.bgWhite
        },
        black: {
            light: '#212121',
            regular: color.black,
        },
        red: {
            regular: color.red,
            bright: color.brightRed
        },
        blue: {
            light: color.lightBlue,
            regular: color.blue,
            bg: '#2196f3'
        },
        grey: {
            light: '#696969',
            regular: color.grey,
            dark: '#444',
            bg: color.bgGrey,
            border: 'rgba(14,19,24,.15)'
        },
        green: {
            light: color.lightGreen,
            regular: color.green,
            dark: color.darkGreen
        },
        success: color.success,
        warning: color.warning,
        danger: color.danger,
        info: color.info,
        link: {
            hover: 'rgba(14,19,24, .07)',
        }
    },
    fontSize: {
        smaller: "font-size: 1.2rem",
        small: "font-size: 1.4rem;",
        base: "font-size: 1.6rem;",
        medium: "font-size: 1.8rem",
        large: "font-size: 2rem",
        h1: "font-size: 7.5rem;",
        h2: "font-size: 5.5rem;",
        h3: "font-size: 3.5rem;",
        h4: "font-size: 2.5rem;",
        button: "font-size: 1.8rem;"
    },
    shadow: {
        text: `text-shadow: .5px .5px 1px ${color.black}`,
        text_links: `1px .75px 1.5px ${color.black}`,
        box: `box-shadow: 0 3px 20px 0 rgba(0, 0, 0, .2)`,
        card: `box-shadow: 1px .5px 3px rgba(0, 0, 0, .15)`,
        intenseBox: `box-shadow: 0 2.5px 10px ${color.black};`,
        filter: `filter: drop-shadow(.15px .15px 1.5px ${color.black});
        filter: drop-shadow(-.15px -.15px 1px ${color.black});`,
        btn: `2px 2px 3px ${color.black}`
    },
    flex: {
        center: 'display: flex; justify-content: center; align-items: center;',
        columnCenter: 'display: flex; flex-direction: column; justify-content: center; align-items: center;',
        column: 'display: flex; flex-direction: column;',

        custom: (justify='start', align='start', direction='row', wrap='no-wrap') => {
            return `
                display: flex; 
                flex-direction: ${direction}
                justify-content: ${justify}; 
                align-items: ${align};
                flex-wrap: ${wrap};
            `
        },
    },
    device: {
        mobileS: `(min-width: ${size.mobileS})`,
        mobileM: `(min-width: ${size.mobileM})`,
        mobileL: `(min-width: ${size.mobileL})`,
        tablet: `(min-width: ${size.tablet})`,
        laptop: `(min-width: ${size.laptop})`,
        laptopL: `(min-width: ${size.laptopL})`,
        desktop: `(min-width: ${size.desktop})`,
        desktopL: `(min-width: ${size.desktop})`
    }
};