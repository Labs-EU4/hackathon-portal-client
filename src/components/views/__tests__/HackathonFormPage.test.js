import React from "react";
import { Router } from "react-router";
import { createMemoryHistory } from "history";
import { Provider } from "react-redux";
import { render } from "@testing-library/react";
import configureStore from "redux-mock-store";
import "@testing-library/jest-dom/extend-expect";
import { initialState } from "../../../utils/mockData";
import HackathonFormPage from "../HackathonFormPage";
import { theme } from "../../../assets/styles/ThemeStyling";
import { ThemeProvider } from "styled-components";

const history = createMemoryHistory();

let mockStore;
let store;
let component;

beforeEach(() => {
  mockStore = configureStore();
  store = mockStore(initialState);

  component = render(
    <Router history={history}>
      <Provider store={store}>
        <ThemeProvider theme={theme}>
          <HackathonFormPage />
        </ThemeProvider>
      </Provider>
    </Router>
  );
});

describe("Component HackathonFormPage.js renders properly", () => {
  it("asserts that the component renders properly", () => {
    expect(component).toMatchSnapshot();
  });
  it("asserts that the first character of the user's email shows up on the menu", () => {
    expect(component.getByText("8")).toBeInTheDocument();
  });
  it("asserts that the Create New Hackathon text node renders properly on the form component", () => {
    expect(component.getByText("Create New Hackathon")).toBeInTheDocument();
  });
});
