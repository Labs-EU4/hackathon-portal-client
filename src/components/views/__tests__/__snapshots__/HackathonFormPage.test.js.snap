// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component HackathonFormPage.js renders properly asserts that the component renders properly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-dnqmqq kueQMu"
      >
        <div
          class="sc-gzVnrw sc-htoDjs eFnBQO"
        >
          <h3
            class="sc-EHOje sc-VigVT cSGDLg"
          >
            Create New Hackathon
          </h3>
          <p
            class="sc-jzJRlG jJZkgl"
            color="primary"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-times fa-w-11 sc-kAzzGY dnwlsD"
              data-icon="times"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 352 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                fill="currentColor"
              />
            </svg>
          </p>
        </div>
        <form
          action="#"
          class="sc-iwsKbI dsCYug"
        >
          <div
            class="sc-gZMcBi YuTYM"
          >
            <div
              class="sc-jTzLTM bxzhAG"
            >
              <label
                class="sc-cSHVUG zmqIo"
                for="event_title"
              >
                Hackathon Title
              </label>
              <input
                class="sc-chPdSV iulMjd"
                display="wide"
                id="event_title"
                name="event_title"
                type="text"
                value=""
              />
              <span
                class="sc-hMqMXs sc-hSdWYo lfRhmD"
              />
            </div>
            <div
              class="sc-jTzLTM bxzhAG"
            >
              <label
                class="sc-cSHVUG zmqIo"
                for="event_description"
              >
                Description
              </label>
              <div
                class="sc-ckVGcZ bkFhs"
              >
                <textarea
                  id="event_description"
                  name="event_description"
                  type="text"
                />
              </div>
              <span
                class="sc-hMqMXs sc-hSdWYo lfRhmD"
              />
            </div>
            <div
              class="sc-jTzLTM bxzhAG"
            >
              <div
                class="sc-fjdhpX gpYlNN"
              >
                <label
                  class="sc-cSHVUG zmqIo"
                  for="start_date"
                >
                  Event Starts
                </label>
                <input
                  class="sc-chPdSV hhxnWJ"
                  id="start_date"
                  min="2020-03-20"
                  name="start_date"
                  placeholder="Event starts"
                  type="date"
                  value="2020-03-20"
                />
                <span
                  class="sc-hMqMXs sc-hSdWYo lfRhmD"
                />
              </div>
              <div
                class="sc-fjdhpX gpYlNN"
              >
                <label
                  class="sc-cSHVUG zmqIo"
                  for="end_date"
                >
                  Event Ends
                </label>
                <input
                  class="sc-chPdSV hhxnWJ"
                  id="end_date"
                  min=""
                  name="end_date"
                  placeholder="Event ends"
                  type="date"
                  value="2020-03-20"
                />
                <span
                  class="sc-hMqMXs sc-hSdWYo lfRhmD"
                />
              </div>
            </div>
            <div
              class="sc-jTzLTM bxzhAG"
              id="grading_rubrics"
            >
              <label
                class="sc-cSHVUG zmqIo"
                for="grading_rubrics"
              >
                Grading Rubrics (tick all that apply)
              </label>
              <p
                class="sc-eNQAEJ ddBNwG"
              >
                Judges will be expected to grade project submissions on which one of the following
              </p>
              Â 
              <div
                class="sc-kGXeez ipfKQQ"
              >
                <input
                  class="sc-dxgOiQ hBWsyH"
                  id="presentation"
                  name="rubrics"
                  type="checkbox"
                  value="presentation"
                />
                <label
                  class="sc-cSHVUG zmqIo"
                  for="presentation"
                >
                  Presentation
                </label>
              </div>
              <div
                class="sc-kGXeez ipfKQQ"
              >
                <input
                  class="sc-dxgOiQ hBWsyH"
                  id="product_fit"
                  name="rubrics"
                  type="checkbox"
                  value="product_fit"
                />
                <label
                  class="sc-cSHVUG zmqIo"
                  for="product_fit"
                >
                  Product Fit
                </label>
              </div>
              <div
                class="sc-kGXeez ipfKQQ"
              >
                <input
                  class="sc-dxgOiQ hBWsyH"
                  id="functionality"
                  name="rubrics"
                  type="checkbox"
                  value="functionality"
                />
                <label
                  class="sc-cSHVUG zmqIo"
                  for="functionality"
                >
                  Functionality
                </label>
              </div>
              <div
                class="sc-kGXeez ipfKQQ"
              >
                <input
                  class="sc-dxgOiQ hBWsyH"
                  id="innovation"
                  name="rubrics"
                  type="checkbox"
                  value="innovation"
                />
                <label
                  class="sc-cSHVUG zmqIo"
                  for="innovation"
                >
                  Innovation
                </label>
              </div>
              <div
                class="sc-kGXeez ipfKQQ"
              >
                <input
                  class="sc-dxgOiQ hBWsyH"
                  id="product_design"
                  name="rubrics"
                  type="checkbox"
                  value="product_design"
                />
                <label
                  class="sc-cSHVUG zmqIo"
                  for="product_design"
                >
                  Product Design
                </label>
              </div>
              <div
                class="sc-kGXeez ipfKQQ"
              >
                <input
                  class="sc-dxgOiQ hBWsyH"
                  id="extensibility"
                  name="rubrics"
                  type="checkbox"
                  value="extensibility"
                />
                <label
                  class="sc-cSHVUG zmqIo"
                  for="extensibility"
                >
                  Extensibility
                </label>
              </div>
            </div>
            <div
              class="sc-jTzLTM bxzhAG"
            >
              <div
                class="sc-fjdhpX gpYlNN"
              >
                <label
                  class="sc-cSHVUG zmqIo"
                  for="participation_type"
                >
                  Participation Type
                </label>
                <select
                  class="sc-jKJlTe ePeoXD"
                  id="participation_type"
                  name="participation_type"
                >
                  <option
                    disabled=""
                    hidden=""
                    value=""
                  >
                    Choose
                  </option>
                  <option
                    value="individual"
                  >
                    Individual
                  </option>
                  <option
                    value="team"
                  >
                    Team
                  </option>
                  <option
                    value="both"
                  >
                    Both
                  </option>
                </select>
                <span
                  class="sc-hMqMXs sc-hSdWYo lfRhmD"
                />
              </div>
              <div
                class="sc-fjdhpX gpYlNN"
              >
                <label
                  class="sc-cSHVUG zmqIo"
                  for="event_category"
                >
                  Event Category
                </label>
                <select
                  class="sc-jKJlTe ePeoXD"
                  id="event_category"
                  name="event_category"
                >
                  <option
                    disabled=""
                    hidden=""
                    value=""
                  >
                    Choose
                  </option>
                  <option
                    value="2"
                  >
                    Summer Hackton
                  </option>
                  <option
                    value="3"
                  >
                    World Hackton
                  </option>
                  <option
                    value="3"
                  >
                    Asia Hackton
                  </option>
                  <option
                    value="3"
                  >
                    Gamer's Hackton
                  </option>
                  <option
                    value="3"
                  >
                    Advanced Hackton
                  </option>
                </select>
                <span
                  class="sc-hMqMXs sc-hSdWYo lfRhmD"
                />
              </div>
            </div>
          </div>
          <div
            class="sc-gZMcBi YuTYM"
          >
            <div
              class="sc-jTzLTM bxzhAG"
            >
              <label
                class="sc-cSHVUG zmqIo"
                for="location"
              >
                Location
              </label>
              <input
                class="sc-chPdSV iulMjd"
                display="wide"
                id="location"
                name="location"
                type="text"
                value=""
              />
              <span
                class="sc-hMqMXs sc-hSdWYo lfRhmD"
              />
            </div>
            <div
              class="sc-jTzLTM bxzhAG"
            >
              <label
                class="sc-cSHVUG zmqIo"
                for="guidelines"
              >
                Guidelines
              </label>
              <div
                class="sc-ckVGcZ bkFhs"
              >
                <textarea
                  id="guidelines"
                  name="guidelines"
                  type="text"
                />
              </div>
              <span
                class="sc-hMqMXs sc-hSdWYo lfRhmD"
              />
            </div>
            <div
              class="sc-jTzLTM bxzhAG"
            >
               
              <label
                class="sc-cSHVUG zmqIo"
                for="input_tags"
              >
                Tags
              </label>
              <div
                class="input-tag"
                id="input_tags"
                tags=""
              >
                <ul
                  class="input-tag__tags"
                >
                  <li
                    class="input-tag__tags__input"
                  >
                    <input
                      maxlength="20"
                      type="text"
                    />
                  </li>
                </ul>
              </div>
            </div>
            <div
              class="sc-jTzLTM bxzhAG"
              id="submission_requirements"
            >
              <label
                class="sc-cSHVUG zmqIo"
                for="submission_requirements"
              >
                Project Submission Requirements (tick all that apply)
              </label>
              <p
                class="sc-eNQAEJ ddBNwG"
              >
                Participants will be expected to submit which one of the following
              </p>
              <div
                class="sc-kGXeez ipfKQQ"
              >
                <input
                  class="sc-dxgOiQ hBWsyH"
                  id="video_url"
                  name="requirements"
                  type="checkbox"
                  value="video_url"
                />
                <label
                  class="sc-cSHVUG zmqIo"
                  for="video_url"
                >
                  Video URL
                </label>
              </div>
              <div
                class="sc-kGXeez ipfKQQ"
              >
                <input
                  class="sc-dxgOiQ hBWsyH"
                  id="github_url"
                  name="requirements"
                  type="checkbox"
                  value="github_url"
                />
                <label
                  class="sc-cSHVUG zmqIo"
                  for="github_url"
                >
                  GitHub URL
                </label>
              </div>
            </div>
            <div
              class="sc-jTzLTM bxzhAG"
            />
            <button
              class="sc-bdVaJa gFEKGJ sc-gqjmRU dEyQNc"
              color="green"
              type="submit"
            >
              Submit
            </button>
          </div>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-dnqmqq kueQMu"
    >
      <div
        class="sc-gzVnrw sc-htoDjs eFnBQO"
      >
        <h3
          class="sc-EHOje sc-VigVT cSGDLg"
        >
          Create New Hackathon
        </h3>
        <p
          class="sc-jzJRlG jJZkgl"
          color="primary"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-times fa-w-11 sc-kAzzGY dnwlsD"
            data-icon="times"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 352 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
              fill="currentColor"
            />
          </svg>
        </p>
      </div>
      <form
        action="#"
        class="sc-iwsKbI dsCYug"
      >
        <div
          class="sc-gZMcBi YuTYM"
        >
          <div
            class="sc-jTzLTM bxzhAG"
          >
            <label
              class="sc-cSHVUG zmqIo"
              for="event_title"
            >
              Hackathon Title
            </label>
            <input
              class="sc-chPdSV iulMjd"
              display="wide"
              id="event_title"
              name="event_title"
              type="text"
              value=""
            />
            <span
              class="sc-hMqMXs sc-hSdWYo lfRhmD"
            />
          </div>
          <div
            class="sc-jTzLTM bxzhAG"
          >
            <label
              class="sc-cSHVUG zmqIo"
              for="event_description"
            >
              Description
            </label>
            <div
              class="sc-ckVGcZ bkFhs"
            >
              <textarea
                id="event_description"
                name="event_description"
                type="text"
              />
            </div>
            <span
              class="sc-hMqMXs sc-hSdWYo lfRhmD"
            />
          </div>
          <div
            class="sc-jTzLTM bxzhAG"
          >
            <div
              class="sc-fjdhpX gpYlNN"
            >
              <label
                class="sc-cSHVUG zmqIo"
                for="start_date"
              >
                Event Starts
              </label>
              <input
                class="sc-chPdSV hhxnWJ"
                id="start_date"
                min="2020-03-20"
                name="start_date"
                placeholder="Event starts"
                type="date"
                value="2020-03-20"
              />
              <span
                class="sc-hMqMXs sc-hSdWYo lfRhmD"
              />
            </div>
            <div
              class="sc-fjdhpX gpYlNN"
            >
              <label
                class="sc-cSHVUG zmqIo"
                for="end_date"
              >
                Event Ends
              </label>
              <input
                class="sc-chPdSV hhxnWJ"
                id="end_date"
                min=""
                name="end_date"
                placeholder="Event ends"
                type="date"
                value="2020-03-20"
              />
              <span
                class="sc-hMqMXs sc-hSdWYo lfRhmD"
              />
            </div>
          </div>
          <div
            class="sc-jTzLTM bxzhAG"
            id="grading_rubrics"
          >
            <label
              class="sc-cSHVUG zmqIo"
              for="grading_rubrics"
            >
              Grading Rubrics (tick all that apply)
            </label>
            <p
              class="sc-eNQAEJ ddBNwG"
            >
              Judges will be expected to grade project submissions on which one of the following
            </p>
            Â 
            <div
              class="sc-kGXeez ipfKQQ"
            >
              <input
                class="sc-dxgOiQ hBWsyH"
                id="presentation"
                name="rubrics"
                type="checkbox"
                value="presentation"
              />
              <label
                class="sc-cSHVUG zmqIo"
                for="presentation"
              >
                Presentation
              </label>
            </div>
            <div
              class="sc-kGXeez ipfKQQ"
            >
              <input
                class="sc-dxgOiQ hBWsyH"
                id="product_fit"
                name="rubrics"
                type="checkbox"
                value="product_fit"
              />
              <label
                class="sc-cSHVUG zmqIo"
                for="product_fit"
              >
                Product Fit
              </label>
            </div>
            <div
              class="sc-kGXeez ipfKQQ"
            >
              <input
                class="sc-dxgOiQ hBWsyH"
                id="functionality"
                name="rubrics"
                type="checkbox"
                value="functionality"
              />
              <label
                class="sc-cSHVUG zmqIo"
                for="functionality"
              >
                Functionality
              </label>
            </div>
            <div
              class="sc-kGXeez ipfKQQ"
            >
              <input
                class="sc-dxgOiQ hBWsyH"
                id="innovation"
                name="rubrics"
                type="checkbox"
                value="innovation"
              />
              <label
                class="sc-cSHVUG zmqIo"
                for="innovation"
              >
                Innovation
              </label>
            </div>
            <div
              class="sc-kGXeez ipfKQQ"
            >
              <input
                class="sc-dxgOiQ hBWsyH"
                id="product_design"
                name="rubrics"
                type="checkbox"
                value="product_design"
              />
              <label
                class="sc-cSHVUG zmqIo"
                for="product_design"
              >
                Product Design
              </label>
            </div>
            <div
              class="sc-kGXeez ipfKQQ"
            >
              <input
                class="sc-dxgOiQ hBWsyH"
                id="extensibility"
                name="rubrics"
                type="checkbox"
                value="extensibility"
              />
              <label
                class="sc-cSHVUG zmqIo"
                for="extensibility"
              >
                Extensibility
              </label>
            </div>
          </div>
          <div
            class="sc-jTzLTM bxzhAG"
          >
            <div
              class="sc-fjdhpX gpYlNN"
            >
              <label
                class="sc-cSHVUG zmqIo"
                for="participation_type"
              >
                Participation Type
              </label>
              <select
                class="sc-jKJlTe ePeoXD"
                id="participation_type"
                name="participation_type"
              >
                <option
                  disabled=""
                  hidden=""
                  value=""
                >
                  Choose
                </option>
                <option
                  value="individual"
                >
                  Individual
                </option>
                <option
                  value="team"
                >
                  Team
                </option>
                <option
                  value="both"
                >
                  Both
                </option>
              </select>
              <span
                class="sc-hMqMXs sc-hSdWYo lfRhmD"
              />
            </div>
            <div
              class="sc-fjdhpX gpYlNN"
            >
              <label
                class="sc-cSHVUG zmqIo"
                for="event_category"
              >
                Event Category
              </label>
              <select
                class="sc-jKJlTe ePeoXD"
                id="event_category"
                name="event_category"
              >
                <option
                  disabled=""
                  hidden=""
                  value=""
                >
                  Choose
                </option>
                <option
                  value="2"
                >
                  Summer Hackton
                </option>
                <option
                  value="3"
                >
                  World Hackton
                </option>
                <option
                  value="3"
                >
                  Asia Hackton
                </option>
                <option
                  value="3"
                >
                  Gamer's Hackton
                </option>
                <option
                  value="3"
                >
                  Advanced Hackton
                </option>
              </select>
              <span
                class="sc-hMqMXs sc-hSdWYo lfRhmD"
              />
            </div>
          </div>
        </div>
        <div
          class="sc-gZMcBi YuTYM"
        >
          <div
            class="sc-jTzLTM bxzhAG"
          >
            <label
              class="sc-cSHVUG zmqIo"
              for="location"
            >
              Location
            </label>
            <input
              class="sc-chPdSV iulMjd"
              display="wide"
              id="location"
              name="location"
              type="text"
              value=""
            />
            <span
              class="sc-hMqMXs sc-hSdWYo lfRhmD"
            />
          </div>
          <div
            class="sc-jTzLTM bxzhAG"
          >
            <label
              class="sc-cSHVUG zmqIo"
              for="guidelines"
            >
              Guidelines
            </label>
            <div
              class="sc-ckVGcZ bkFhs"
            >
              <textarea
                id="guidelines"
                name="guidelines"
                type="text"
              />
            </div>
            <span
              class="sc-hMqMXs sc-hSdWYo lfRhmD"
            />
          </div>
          <div
            class="sc-jTzLTM bxzhAG"
          >
             
            <label
              class="sc-cSHVUG zmqIo"
              for="input_tags"
            >
              Tags
            </label>
            <div
              class="input-tag"
              id="input_tags"
              tags=""
            >
              <ul
                class="input-tag__tags"
              >
                <li
                  class="input-tag__tags__input"
                >
                  <input
                    maxlength="20"
                    type="text"
                  />
                </li>
              </ul>
            </div>
          </div>
          <div
            class="sc-jTzLTM bxzhAG"
            id="submission_requirements"
          >
            <label
              class="sc-cSHVUG zmqIo"
              for="submission_requirements"
            >
              Project Submission Requirements (tick all that apply)
            </label>
            <p
              class="sc-eNQAEJ ddBNwG"
            >
              Participants will be expected to submit which one of the following
            </p>
            <div
              class="sc-kGXeez ipfKQQ"
            >
              <input
                class="sc-dxgOiQ hBWsyH"
                id="video_url"
                name="requirements"
                type="checkbox"
                value="video_url"
              />
              <label
                class="sc-cSHVUG zmqIo"
                for="video_url"
              >
                Video URL
              </label>
            </div>
            <div
              class="sc-kGXeez ipfKQQ"
            >
              <input
                class="sc-dxgOiQ hBWsyH"
                id="github_url"
                name="requirements"
                type="checkbox"
                value="github_url"
              />
              <label
                class="sc-cSHVUG zmqIo"
                for="github_url"
              >
                GitHub URL
              </label>
            </div>
          </div>
          <div
            class="sc-jTzLTM bxzhAG"
          />
          <button
            class="sc-bdVaJa gFEKGJ sc-gqjmRU dEyQNc"
            color="green"
            type="submit"
          >
            Submit
          </button>
        </div>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
